### PR TITLE
Adds lightWallet Seed Logic and View Seed Modal 

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 1]
```diff
@@ -44,23 +55,111 @@ class BackupSeedContainer extends React.Component<IDispatchProps, IComponentStat
...
+        <BackupSeedContainerComponent
+          verifyBackupPhrase={this.props.verifyBackupPhrase}
+          seed={this.props.seed}
+        />
+      );
+    this.props.fetchSeed();
+    return <LoadingIndicator />;
+  }
+
+  lockedWalletComponent = (): React.ReactNode => (
```
Comment (krzkaczor) [inline app/components/settings/backupSeed/BackupSeed.tsx]: here you dont need `something = () => ` syntax. Use the same as above so `lockedWalletComponent(): React.ReactNode {....`

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 2]
```diff
@@ -44,23 +55,111 @@ class BackupSeedContainer extends React.Component<IDispatchProps, IComponentStat
...
+  }
+  unlockedWalletComponent(): React.ReactNode {
+    if (this.props.seed)
+      return (
+        <BackupSeedContainerComponent
+          verifyBackupPhrase={this.props.verifyBackupPhrase}
+          seed={this.props.seed}
+        />
+      );
+    this.props.fetchSeed();
```
Comment (krzkaczor) [inline app/components/settings/backupSeed/BackupSeed.tsx]: this is bad — we shouldnt issue actions in random places in render methods

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 3]
```diff
@@ -44,23 +55,111 @@ class BackupSeedContainer extends React.Component<IDispatchProps, IComponentStat
...
+  render(): React.ReactNode {
+    return this.props.isUnlocked ? (
+      <>{this.unlockedWalletComponent()}</>
+    ) : (
+      <>{this.lockedWalletComponent()}</>
+    );
+  }
+}
+
 export const BackupSeed = appConnect<{}, IDispatchProps>({
```
Comment (krzkaczor) [inline app/components/settings/backupSeed/BackupSeed.tsx]: why {} here when it takes some states props?

File: app/components/modals/viewSeedModal.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  clearSeed: () => void;
+}
+
+interface IState {
+  pageNo: number;
+  seed?: string[];
+}
+
+export class ViewSeedModalComponent extends React.Component<IStateProps & IDispatchProps, IState> {
+  constructor(props: any) {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/modals/viewSeedModal.tsx]: good catch

File: app/components/modals/viewSeedModal.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    super(props);
+    this.props.fetchSeed();
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/modals/viewSeedModal.tsx]: I didn't want to put it in `componentWillMount` on purpose as there are cases where the component hangs if anything wrong happens with fetch, but now that I think of it that is only for async functions where you have to `await`  
Changed

File: app/components/modals/viewSeedModal.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  render(): React.ReactNode {
+    return this.props.seed ? (
+      <BackupSeedDisplayComponent totalSteps={2} startingStep={1} words={this.props.seed} isModal />
+    ) : (
+      <LoadingIndicator />
+    );
+  }
+}
+
+const Test: React.SFC<any> = props => {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/modals/viewSeedModal.tsx]: this shouldn't be named `test` 

Changed to `ViewSeedModalComponent`

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 7]
```diff
@@ -1,25 +1,36 @@
...
+import * as cn from "classnames";
+import { LoadingIndicator } from "../../shared/LoadingIndicator";
+import * as styles from "./BackupSeed.module.scss";
 
-interface IDispatchProps {
+interface IComponentDisptachProps {
   verifyBackupPhrase: () => void;
 }
 
+interface IComponentProps {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: done

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 8]
```diff
@@ -44,23 +55,111 @@ class BackupSeedContainer extends React.Component<IDispatchProps, IComponentStat
       case 1:
         return <BackupSeedIntro onBack={appRoutes.settings} onNext={this.onNext} />;
       case 2:
-        return <BackupSeedDisplay onBack={this.onBack} onNext={this.onNext} words={words} />;
+        return (
+          <BackupSeedDisplay
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: Hmm i have no strong opinions regarding this.

File: app/components/settings/backupSeed/BackupSeed.tsx [hunk 9]
```diff
@@ -44,23 +55,111 @@ class BackupSeedContainer extends React.Component<IDispatchProps, IComponentStat
...
+  IDispatchProps & IStateProps,
+  IBackupSeedContainerState
+> {
+  constructor(props: IDispatchProps & IStateProps) {
+    super(props);
+  }
+  componentWillUnmount(): void {
+    this.props.clearSeed();
+  }
+  unlockedWalletComponent(): React.ReactNode {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: i was thinking about how they should be named

File: app/components/settings/backupSeed/BackupSeed.tsx (hunk 1 above)
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: done

File: app/components/settings/backupSeed/BackupSeed.tsx (hunk 2 above)
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: this is a bit more complicated problem, i will leave it like this for now until we find a general solution, 
Take into note the complicated nature of Modals and all the react functions, i tested with a couple of them and nothing seems to get triggered correctly.

File: app/components/settings/backupSeed/BackupSeed.tsx (hunk 3 above)
Comment (Moejoe90) [reply app/components/settings/backupSeed/BackupSeed.tsx]: you are correct

File: app/components/modals/LightWalletSign.tsx [hunk 10]
```diff
@@ -13,16 +12,13 @@ interface IStateProps {
   isUnlocked: boolean;
 }
 
-interface IDispatchProps {
+interface IProps {
```
Comment (krzkaczor) [inline app/components/modals/LightWalletSign.tsx]: this should be IOwnProps

File: app/components/modals/LightWalletSign.tsx [hunk 11]
```diff
@@ -59,12 +54,8 @@ export class LightWalletSignPromptComponent extends React.Component<
   }
 }
 
-export const LightWalletSignPrompt = appConnect<IStateProps, IDispatchProps>({
+export const LightWalletSignPrompt = appConnect<IStateProps>({
```
Comment (krzkaczor) [inline app/components/modals/LightWalletSign.tsx]: here pass own props as 3rd arg

File: app/components/modals/viewSeedModal.tsx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+          />
+        </Col>
+      </Row>
+    ) : (
+      <LoadingIndicator />
+    );
+  }
+}
+
+const ViewSeedModalComponent: React.SFC<any> = props => {
```
Comment (krzkaczor) [inline app/components/modals/viewSeedModal.tsx]: any :(

Comment (Moejoe90) [reply app/components/modals/viewSeedModal.tsx]: already solved

File: app/components/modals/LightWalletSign.tsx (hunk 11 above)
Comment (Moejoe90) [reply app/components/modals/LightWalletSign.tsx]: done

File: app/components/modals/LightWalletSign.tsx (hunk 10 above)
Comment (Moejoe90) [reply app/components/modals/LightWalletSign.tsx]: done

**Pull request merged**